Ds3/ResponseParsers/GetSystemInformationParser.cs

```diff
@@ -0,0 +1,69 @@
+﻿/*
+ * ******************************************************************************
+ *   Copyright 2014 Spectra Logic Corporation. All Rights Reserved.
+ *   Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+ *   this file except in compliance with the License. A copy of the License is located at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file.
+ *   This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ *   CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ *   specific language governing permissions and limitations under the License.
+ * ****************************************************************************
+ */
+
+using System;
+using System.IO;
+using System.Net;
+using Ds3.Calls;
+using Ds3.Models;
+using Ds3.Runtime;
+
+/* <Data>
+ *  <ApiVersion>91C76B3B5B01A306A0DFA94C9EE3549A.C3E40C732B27EFA6D25AB24D3E01E851</ApiVersion>
+ *  <BuildInformation>
+ *      <Branch/>
+ *      <Revision/>
+ *      <Version/>
+ *  </BuildInformation>
+ *  <SerialNumber>UNKNOWN</SerialNumber>
+ * </Data>
+ */
+namespace Ds3.ResponseParsers
+{
+    class GetSystemInformationParser : IResponseParser<GetSystemInformationRequest, GetSystemInformationResponse>
+    {
+        public GetSystemInformationResponse Parse(GetSystemInformationRequest request, IWebResponse response)
+        {
+            using (response)
+            {
+                ResponseParseUtilities.HandleStatusCode(response, HttpStatusCode.OK);
+                using (Stream content = response.GetResponseStream())
+                {
+                    var root = XmlExtensions.ReadDocument(content).ElementOrThrow("Data");
+
+                    var buildParams = root.Element("BuildInformation");
+                    string BuildBranch = buildParams.TextOf("Branch");
+                    string BuildRev = buildParams.TextOf("Revision");
+                    string BuildVersion = buildParams.TextOf("Version");
+                    string ApiVersion = root.TextOf("ApiVersion");
+                    string MC5Major = string.Empty;
+                    string MC5Full = string.Empty;
+                    if (!string.IsNullOrEmpty(ApiVersion))
+                    {
+                        string[] mc5s = ApiVersion.Split('.');

```

Comment:
You’re referring to the split of API version into the two MC5’s – or the whole parser?

jk

From: Ryan Moore [mailto:notifications@github.com]
Sent: Tuesday, September 08, 2015 5:23 PM
To: SpectraLogic/ds3_net_sdk <ds3_net_sdk@noreply.github.com>
Cc: John Kranz <johnk@spectralogic.com>
Subject: Re: [ds3_net_sdk] jk -- Add GetSystemInfo (NETSDK-41) and VerifySystemHealth (NETSDK-40… (#46)


In Ds3/ResponseParsers/GetSystemInformationParser.cs<https://github.com/SpectraLogic/ds3_net_sdk/pull/46#discussion_r38992224>:

> +            {

> +                ResponseParseUtilities.HandleStatusCode(response, HttpStatusCode.OK);

> +                using (Stream content = response.GetResponseStream())

> +                {

> +                    var root = XmlExtensions.ReadDocument(content).ElementOrThrow("Data");

> +

> +                    var buildParams = root.Element("BuildInformation");

> +                    string BuildBranch = buildParams.TextOf("Branch");

> +                    string BuildRev = buildParams.TextOf("Revision");

> +                    string BuildVersion = buildParams.TextOf("Version");

> +                    string ApiVersion = root.TextOf("ApiVersion");

> +                    string MC5Major = string.Empty;

> +                    string MC5Full = string.Empty;

> +                    if (!string.IsNullOrEmpty(ApiVersion))

> +                    {

> +                        string[] mc5s = ApiVersion.Split('.');

I don't know if this is needed. We've not done it in the other SDKs.

—
Reply to this email directly or view it on GitHub<https://github.com/SpectraLogic/ds3_net_sdk/pull/46/files#r38992224>.